OpenRA.Game/Map/CellLayer.cs

```diff
@@ -18,13 +18,13 @@ namespace OpenRA
 	// Represents a layer of "something" that covers the map
 	public class CellLayer<T> : IEnumerable<T>
 	{
-		public readonly Size Size;
-		public readonly TileShape Shape;
+		public Size Size { get; private set; }
```

Comment:
I prefer not to do so. The get private set is the most similar to a readonly for public fields. Public fields cannot be interfaced and cannot be mocked in unit tests if I ever decide to do so.

Please, refer to a comment from Jon Skeet for more information (Jon Skeet is a reference in SO): http://stackoverflow.com/questions/8485402/how-to-mock-a-public-readonly-field